### PR TITLE
Rdp cache hunting script

### PR DESCRIPTION
**Pull request by maimorag**
## Status
- [x] In Progress

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-5364)

## Description
As part of the RDP cache hunting use case, a script is needed to pre-process an image file from context. 
The script should receive inputs for the actions to perform on the image, image_size to resize the output image, and the file entry ID. 

## Minimum version of Cortex XSOAR
- [x] 6.5.0

### CODE REVIEW

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+This script pre-process an image file from context.
+
+## Script Data
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md:3]: Generate the readme with example

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+| Script Type | python3 |
+| Cortex XSOAR Version | 6.0.0 |
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| action | The action to perform on the image. |
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md:18]: Add here the possible options as well

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+- auto: PREDEFINED
+  description: The action to perform on the image.
+  name: action
+  predefined:
+  - grayscale
+  - sharpened
+  - original
+  required: true
+- default: false
+  description: The desired width for resizing the image to.
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml:11]: Give example

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  isArray: false
+  name: image_resize_width
+  required: false
+  secret: false
+- default: false
+  description: The desired height for resizing the image to.
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml:17]: give example

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  name: image_resize_height
+  required: false
+  secret: false
+- default: false
+  description: The entryID of the file to process.
+  isArray: false
+  name: file_entry_id
+  required: true
+  secret: false
+comment: This script pre-process an image file from context.
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml:28]: Need a better script description here

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    stream_orig.seek(0)
+    file_result = fileResult(filename=f'original_{name}', data=stream_orig.read())
+    return file_result
+
+
+def get_file_details(entry_id: str) -> tuple[Any, str]:
+    """
+    Generate sharpened image.
+    Args:
+        client (Client): Client to perform calls to GitLab services.
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py:107]: You don't pass any client here
also specify the args you giving for each command

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+        FileResult (dict[str, Any]).
+    """
+    args = demisto.args()
+    entry_id = args.get('file_entry_id')
+    width = arg_to_number(args.get('image_resize_width'))
+    height = arg_to_number(args.get('image_resize_height'))
+    image, name = get_file_details(entry_id)
+    stream_orig = io.BytesIO()
+    final_orig_image = Image.fromarray(image)
+    if width and height:
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py:95]: what will happen if I give only one of the args?

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+    """
+    kernel = np.array([[-1, -1, -1], [-1, 8, -1], [-1, -1, 0]], np.float32)
+    kernel = 1 / 3 * kernel
+    sharp = cv2.filter2D(image, -1, kernel)  # pylint: disable=E1101
+    return sharp
+
+
+def image_resize(image: Image, width: int, height: int):
+    """
+    REsizes image according to width and heights values.
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py:26]:
```suggestion
    Resizes image according to width and height values.
```

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+    Returns:
+        (np.ndarray).
+    """
+    kernel = np.array([[-1, -1, -1], [-1, 8, -1], [-1, -1, 0]], np.float32)
+    kernel = 1 / 3 * kernel
+    sharp = cv2.filter2D(image, -1, kernel)  # pylint: disable=E1101
+    return sharp
+
+
+def image_resize(image: Image, width: int, height: int):
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py:24]: add return type

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+    Generate grayscale image.
+    Returns:
+        FileResult (dict[str, Any]).
+    """
+    args = demisto.args()
+    entry_id = args.get('file_entry_id')
+    width = arg_to_number(args.get('image_resize_width'))
+    height = arg_to_number(args.get('image_resize_height'))
+    image, name = get_file_details(entry_id)
+    stream_gray = io.BytesIO()
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.py:49]: It seems that you're using the same code over and over in all functions. 
Please move it to a helper function and pass to it the relevant stuff for each command

File: Packs/RDPCacheHunting/README.md [hunk 11]
```diff
@@ -0,0 +1,2 @@
+RDP cache hunting use case.
```
[the remark this answers is not in the document]
Comment (maimorag) [reply Packs/RDPCacheHunting/README.md:1]: I'll leave it empty

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+        "SHA256": "<SHA256Example>",
+        "SHA512": "<SHA512Example>",
+        "SSDeep": "24576:<SSDeepExample>",
+        "Size": 1757075,
+        "Type": "JPEG image data, JFIF standard 1.01, aspect ratio, density 1x1, segment length 16, baseline, precision 8, 3024x4032, components 3"
+    }
+}
+```
+
+### Example command
```
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md:85]: 2 examples are enough, you can remove the rest

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/test_data/test_picture_png.png
Comment (Shellyber) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/test_data/test_picture_png.png:1]: Let's use a smaller sized images

**Review by Shellyber — COMMENTED** [review]
Looks wayyy better! nice job!
Please schedule a small demo on Sunday

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+  name: image_resize_height
+  required: false
+  secret: false
+- default: false
+  description: The entryID of the file to process.
+  isArray: false
+  name: file_entry_id
+  required: true
+  secret: false
+comment: This script pre-process(resize, sharpen and grayscale ) an image file from context, given an entry_id.
```
Comment (ShirleyDenkberg) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml:28]:
```suggestion
comment: This script pre-processes (resizes, sharpens, and grayscales) an image file from context, given an entry_id.
```

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+  id: PreProcessImage
+  version: -1
+enabled: true
+name: PreProcessImage
+outputs:
+- contextPath: PreProcessImage.file_entry_id_new
+  description: The entryID of the created file.
+  type: String
+- contextPath: PreProcessImage.action
+  description: The action that were performed,.
```
Comment (ShirleyDenkberg) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/PreProcessImage.yml:39]:
```suggestion
  description: The action that was performed.
```

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
+This script pre-process an image file by preforming one of the actions:
```
Comment (ShirleyDenkberg) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md:1]:
```suggestion
This script pre-processes an image file by preforming one of the following actions:
```

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+- original - return the original image.
+- sharpen - process that emphasizes the edges in an image to make it appear clearer and more focused.
+- grayscale - convert the image to black, white, and gray colors, in which gray has multiple levels (the value of each pixel represents only the intensity information of the light).
+
+Each action can also resize the image when given width and height.
```
Comment (ShirleyDenkberg) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md:6]:
```suggestion
Each action can also resize the image when given the width and height.
```

File: Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+| file_entry_id | The entryID of the file to process. |
+
+## Outputs
+
+---
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| PreProcessImage.file_entry_id_new | The entryID of the created file. | String |
+| PreProcessImage.action | The action that were performed,. | String |
```
Comment (ShirleyDenkberg) [inline Packs/RDPCacheHunting/Scripts/PreProcessImage/README.md:35]:
```suggestion
| PreProcessImage.action | The action that was performed. | String |
```

**Comment by ShirleyDenkberg** [comment]
@maimorag Do we need a pack readme?
@Shellyber Doc review completed.

**Review by Shellyber — APPROVED** [review]

**Pull request merged**